spikeextractors/extractors/cellexplorersortingextractor/cellexplorersortingextractor.py

```diff
@@ -0,0 +1,102 @@
+from spikeextractors import SortingExtractor
+import numpy as np
+from pathlib import Path
+from spikeextractors.extraction_tools import check_valid_unit_id
+from typing import Union
+from scipy.io import loadmat, savemat
+
+PathType = Union[str, Path, None]
+
+
+class CellExplorerSortingExtractor(SortingExtractor):
+    """
+    Extracts spiking information from .mat files stored in the CellExplorer format.
+
+    Spike times are stored in units of seconds.
+
+    Parameters
+    ----------
+    spikes_matfile_path : PathType
+        Path to the sorting_id.spikes.cellinfo.mat file.
+    """
+
+    extractor_name = "CellExplorerSortingExtractor"
+    installed = True
+    is_writable = True
+    mode = "file"
+    installation_mesg = ""
+
+    def __init__(self, spikes_matfile_path: PathType):
+        SortingExtractor.__init__(self)
+
+        spikes_matfile_path = Path(spikes_matfile_path)
+        assert spikes_matfile_path.is_file(), f"The spikes_matfile_path ({spikes_matfile_path}) must exist!"
+        folder_path = spikes_matfile_path.parent
+        sorting_id = spikes_matfile_path.name.split(".")[0]
+        session_info_matfile_path = folder_path / f"{sorting_id}.sessionInfo.mat"
+        assert session_info_matfile_path.is_file(), "No sessionInfo.mat file found in the folder!"
+        session_info_mat = loadmat(file_name=str(session_info_matfile_path.absolute()))
+        assert session_info_mat['sessionInfo']['rates'][0][0]['wideband'], "The sesssionInfo.mat file must contain " \
+            "a 'sessionInfo' struct with field 'rates' containing field 'wideband'!"
+        sampling_frequency = float(
+            session_info_mat['sessionInfo']['rates'][0][0]['wideband'][0][0][0][0]
+        )  # careful not to confuse it with the lfpsamplingrate; reported in units Hz
+        self._sampling_frequency = sampling_frequency
+
+        spikes_mat = loadmat(file_name=str(spikes_matfile_path.absolute()))
+        assert len(set(['UID', 'times']) - set(spikes_mat['spikes'].dtype.names)) == 0, \
+            "The spikes.cellinfo.mat file must contain a 'spikes' struct with fields 'UID' and 'times'!"
+
+        self._unit_ids = list(map(int, spikes_mat['spikes']['UID'][0][0][0]))
+        # CellExplorer reports spike times in units seconds; SpikeExtractors uses time units of sampling frames
+        self._spiketrains = [
+            (np.array([y[0] for y in x]) * sampling_frequency).round().astype(int)
```

Comment:
```suggestion
            (np.array([y[0] for y in x]) * sampling_frequency).astype(int)
```

Is the rounding necessary? casting to `int` should do this for you automatically

Reply:
It is unfortunately; casting to `int` clips the remainder by taking the floor and that in conjunction with the sampling frequency conversion actually produces the occasional value error during testing.

Reply:
ok, this is a good solution then. Would you mind adding this in the code comments?

Reply:
Sure thing!